### PR TITLE
Clean-up of some Twig functions and filters.

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

File: src/DataFixtures/ContentFixtures.php [hunk 1]
```diff
@@ -172,10 +172,12 @@ private function getPresetRecords(): array
             'slug' => 'This is a record in the "Entries" ContentType',
         ];
         $records['blocks'][] = [
-            'title' => 'About',
+            'title' => 'About This Site',
+            'slug' => 'about',
         ];
         $records['blocks'][] = [
             'title' => 'Search',
+            'slug' => 'Search',
```
Comment (JarJak) [inline src/DataFixtures/ContentFixtures.php]: lowercase

File: src/Twig/RecordExtension.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public function getFunctions()
 
         return [
             new TwigFunction('excerpt', [$this, 'excerpt'], $safe),
-            new TwigFunction('listtemplates', [$this, 'dummy']),
+            new TwigFunction('listtemplates', [$this, 'listtemplates']),
             new TwigFunction('pager', [$this, 'pager'], $env + $safe),
             new TwigFunction('menu', [$this, 'menu'], $env + $safe),
             new TwigFunction('sidebarmenu', [$this, 'sidebarmenu']),
```
Comment (JarJak) [inline src/Twig/RecordExtension.php]: sidebar_menu

File: src/Twig/RecordExtension.php [hunk 3]
```diff
@@ -53,7 +53,7 @@ public function getFunctions()
 
         return [
             new TwigFunction('excerpt', [$this, 'excerpt'], $safe),
-            new TwigFunction('listtemplates', [$this, 'dummy']),
+            new TwigFunction('listtemplates', [$this, 'listtemplates']),
```
Comment (JarJak) [inline src/Twig/RecordExtension.php]: list_templates

File: src/Twig/RecordExtension.php [hunk 4]
```diff
@@ -65,14 +65,9 @@ public function getFunctions()
         ];
     }
 
-    public function dummy($input = null)
+    public function listtemplates()
```
Comment (JarJak) [inline src/Twig/RecordExtension.php]: getTemplatesList(): array

File: src/Twig/LocaleExtension.php [hunk 5]
```diff
@@ -426,4 +439,35 @@ private function getFlagCodes()
             'ZW' => 'Zimbabwe',
         ]);
     }
+
+    public function localedatetime($dateTime, $format = '%B %e, %Y %H:%M', $locale = 0)
```
Comment (JarJak) [inline src/Twig/LocaleExtension.php]: types

File: src/Twig/TextExtension.php [hunk 6]
```diff
@@ -35,8 +35,12 @@ public function slug($str): string
         return Str::slug((string) $str);
     }
 
-    public function dummy($input = null)
+    public function ucwords($content, string $delimiters = ''): string
```
Comment (JarJak) [inline src/Twig/TextExtension.php]: $content -> string $string

File: templates/_base/layout.html.twig [hunk 7]
```diff
@@ -30,7 +30,7 @@
             site-name="{% if config is defined %}{{ config.get('general/sitename') }}{% endif %}"
             :menu="{{ sidebarmenu() }}"
             user="{{ user|default }}"
-            :labels="{{ labels|json_decode }}"
+            :labels="{{ labels }}"
```
Comment (JarJak) [inline templates/_base/layout.html.twig]: Actually, what are labels?

File: templates/content/_fields.html.twig [hunk 8]
```diff
@@ -1,12 +1,13 @@
...
+
     <div class="tab-pane {% if loop.first %}show active{% endif %}" id="{{ group|slug }}" role="tabpanel"
          aria-labelledby="{{ group|slug }}-tab">
         {% for key, fielddefinition in record.definition.fields if fielddefinition.group == group %}
 
             {% if record.hasField(key) %}
                 {% set field = record.getField(key)|translated(currentlocale) %}
             {% else %}
-                {% set field = field_factory(key, fielddefinition) %}
+                {% set field = record|field_factory(key, fielddefinition) %}
```
Comment (JarJak) [inline templates/content/_fields.html.twig]: it should be a function, not filter

File: src/Twig/LocaleExtension.php (hunk 5 above)
Comment (bobdenotter) [reply src/Twig/LocaleExtension.php]: Added them, except for `$dateTime`, which is either `string` or `\DateTime`

File: templates/_base/layout.html.twig (hunk 7 above)
Comment (bobdenotter) [reply templates/_base/layout.html.twig]: See a few lines above it.. It injects labels provided by Bolt, in the correct locale. 

```
{% set labels = jsonlabels(['about.bolt_documentation', 'action.visit_site', 'action.create_new', ['general.greeting', {'%name%': user_display_name}], 'action.logout', 'action.edit_profile']) %}
```

Comment (JarJak) [reply templates/_base/layout.html.twig]: so it is same as this?
```
[
	'about.bolt_documentation'|trans, 
	'action.visit_site'|trans, 
	'action.create_new'|trans, 
	'general.greeting'|trans({'%name%': user_display_name}), 
	'action.logout', 
	'action.edit_profile'
]|json_encode
```

File: src/DataFixtures/ContentFixtures.php (hunk 1 above)
Comment (bobdenotter) [reply src/DataFixtures/ContentFixtures.php]: done

File: src/Twig/RecordExtension.php (hunk 3 above)
Comment (bobdenotter) [reply src/Twig/RecordExtension.php]: done

File: src/Twig/TextExtension.php (hunk 6 above)
Comment (bobdenotter) [reply src/Twig/TextExtension.php]: done

File: templates/content/_fields.html.twig (hunk 8 above)
Comment (bobdenotter) [reply templates/content/_fields.html.twig]: Reverted this bit. It's it's own function again.

File: src/Twig/LocaleExtension.php [hunk 9]
```diff
@@ -426,4 +439,35 @@ private function getFlagCodes()
             'ZW' => 'Zimbabwe',
         ]);
     }
+
+    public function localedatetime($dateTime, string $format = '%B %e, %Y %H:%M', ?string $locale = '0')
```
Comment (JarJak) [inline src/Twig/LocaleExtension.php]: return type

Comment (JarJak) [reply src/Twig/LocaleExtension.php]: add phpdoc what is accepted as $dateTime

File: src/Twig/RecordExtension.php (hunk 2 above)
Comment (bobdenotter) [reply src/Twig/RecordExtension.php]: done

File: src/Twig/RecordExtension.php (hunk 4 above)
Comment (bobdenotter) [reply src/Twig/RecordExtension.php]: done

File: templates/_base/layout.html.twig (hunk 7 above)
Comment (bobdenotter) [reply templates/_base/layout.html.twig]: Yeah, pretty much.

File: src/Twig/LocaleExtension.php (hunk 9 above)
Comment (bobdenotter) [reply src/Twig/LocaleExtension.php]: done.

**Comment by bobdenotter** [comment]
Rebased, and I even fixed the merge conflict. ;-)

File: templates/_base/layout.html.twig (hunk 7 above)
Comment (JarJak) [reply templates/_base/layout.html.twig]: so function can be removed or at least renamed? there is nothing special about labels

**Comment by bobdenotter** [comment]
Sometimes I can not anticipate whether you feel something is _"No logic in templates! This should be a function!"_ and when it is _"This function is unneeded, replace it with some Twig!"_ ;-)

In this case, i kinda agree that this is perfectly fine to do in Twig, so I've removed `jsonlabels`. It's a leftover from my very first Bolt 4 scaffolding.

**Comment by JarJak** [comment]
>Sometimes I can not anticipate whether you feel something is "No logic in templates! This should be a function!" and when it is "This function is unneeded, replace it with some Twig!"

if something contains logic that is worth unit testing -> function
if something is too simple to test (simple instructions, transitions that can be done with built-in filters) AND is useful only in Twig (and not in e.g. API) -> can be in template

**Pull request merged**